cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go

```diff
@@ -300,7 +308,13 @@ func runAgent(ctx context.Context,
 
 	logger.SetupLogging(logConfig)
 	log.Printf("I! Starting AmazonCloudWatchAgent %s", agentinfo.Version())
-
+	// Need to set SDK log level before plugins get loaded.
+	// Some aws.Config objects get created early and live forever which means
+	// we cannot change the sdk og level without restarting the Agent.
+	// For example CloudWatch.Connect().
+	sdkLogLevel := os.Getenv(envconfig.AWS_SDK_LOG_LEVEL)
+	configaws.SetSDKLogLevel(wlog.LogLevel(), sdkLogLevel)
+	log.Printf("I! AWS SDK log level, %s", sdkLogLevel)
```

Comment:
At first I just relied on `reloadLoop()` to call `configaws.SetSDKLogLevel()` when it it detected the `env-config.json` file changed. However that wouldn't happen for some number of seconds after the Agent had started. Most importantly, that was happening AFTER the output plugins had been loaded.

So adding the code here is necessary.

Reply:
We had that other issue for allowing log level changes without restarting the agent. The implication here is that this change doesn't allow us to turn on/off the debug level without restarting the agent, right?

Reply:
This PR does not let you turn on/off AWS SDK debugging without restarting the Agent.
Even though the `aws.Config.LogLevel` is a pointer to an integer, changing it after the `aws.Config` object has been instantiated had no effect. And the CloudWatch and CloudWatchLogs plugins DO NOT create a new `aws.Config` object each time they do an upload.

You can change the general log level of the CloudWatch Agent at runtime and it will take effect the next time `reloadLoop()` reads the `env-config.json`.

Reply:
I think it's reasonable. Just figured I'd mention it since I noticed it.

Reply:
Just noticed - if we print out `sdkLogLevel` and `os.Getenv` doesn't find anything for the key, will the log still be useful?

Reply:
It was useful for me to know that the SDK Log level is not set (empty string).
As with all logging it is subjective.
I have no issues with removing the line all together.
I felt it was okay to leave in because it only logs once when the Agent starts.

I could change it to:
```
log.Printf("I! AWS SDK log level, %d", *configaws.SDKLogLevel())
```
But that is not very readable since you then need to look up the integer value of the matching ENUM.
Since this log message only occurs once when the Agent starts I felt it was okay to leave in.

What do you prefer?

Reply:
1 suggestion:
```
	if sdkLogLevel == "" {
		log.Printf("I! AWS SDK log level not set")
	} else {
		log.Printf("I! AWS SDK log level, %s", sdkLogLevel)
	}
```

Reply:
I think it's more obvious when the SDK log level isn't set with your suggested change, and I'd prefer that. Is it possible for someone to set the log level string as non-empty whitespace characters, such that we wouldn't know there's an issue just by printing it out? I guess that's where my head was at when I asked about this.